print-802_11.c

```diff
@@ -1329,6 +1347,30 @@ parse_elements(netdissect_options *ndo,
 				pbody->tim_present = 1;
 			}
 			break;
+		case E_MESHID:
+			memcpy(&meshid, p + offset, 2);
```

Comment:
Should print-802_11.c just use `GET_CPY_BYTES()` instead of `memcpy()`?

Reply:
I think that it is not strictly necessary, but it would make it easier.
As for your previous comment about the structure being overlaid, the memcpy() is doing exactly that.

Reply:
> As for your previous comment about the structure being overlaid, the memcpy() is doing exactly that.

Not exactly.

The first memcpy - which is done for every IE, not just for the one - is copying the first two bytes of the IE, which are the type and length.  If padding is inserted between 1-byte fields, We Have A Problem with just about every data structure we have, but I know of no compilers that would do that.

The second memcpy - which is also done from most if not all Its - is copying into the string buffer in the structure, not into the structure as a whole.

This is the way other IEs are handled as well.

It might be cleaner if the structures in question didn't have an `element_id` member, as that's not used (the type of the structure implies which IE it is), at which point there would be nothing that even looks like overlaying being done.

Reply:
> It might be cleaner if the structures in question didn't have an element_id member, as that's not used (the type of the structure implies which IE it is), at which point there would be nothing that even looks like overlaying being done.

I've checked in that change as 33b6502c99917c20a81303e2dc60bc924bb4b0c5, with further improvements in 36fded60133928c151288975be475d2164230cec.

`struct meshid_t` should be changed to match, with the `element_id` element removed, and with the `length` element being a `u_int`.

The first line of the `case E_MESHID:` code should be changed from a `memcpy()` to

    meshid.length = elementlen;

Reply:
@mcr & @guyharris , thanks for the review. As suggested, i have made the changes and updated the PR.

And as you both might have seen already, the newly added MESHID field is printed to stdout enclosed by a parenthesis like below. Reason for introducing this parenthesis is, earlier i was printing the MESHID immediately after the empty parenthesis of the SSID field so also enclosed the MESHID field with it for better reability. But then i have moved it to the end of the line, because if someone have written a tcpdump stdout parser the newly added field at the middle of the line would break their logic.
```
12:20:37.867811 9526800862us tsft 6.0 Mb/s 5745 MHz 11a -34dBm signal [bit 22] Beacon () [6.0* 9.0 12.0* 18.0 24.0* 36.0 48.0 54.0 Mbit] IBSS CH: 149, PRIVACY (MESHID: 11s-mesh-network)
```  
Now i have some second thoughts. Do you suggest keeping the MESHID field as it is with a parenthesis, or introduce a comma before printing the MESHID without any parenthesis like below ?
```
12:20:37.867811 9526800862us tsft 6.0 Mb/s 5745 MHz 11a -34dBm signal [bit 22] Beacon () [6.0* 9.0 12.0* 18.0 24.0* 36.0 48.0 54.0 Mbit] IBSS CH: 149, PRIVACY, MESHID: 11s-mesh-network
```

Reply:
@mcr, @guyharris, @fxlb, I request you to suggest one of the two ways mentioned in my previous comment to print the "meshid" to stdout, so that I can update the PR if needed. Thanks.
1. Using parenthesis - the commit in this Pull Request uses this format to print "Mesh ID"
    `
12:20:37.867811 9526800862us tsft 6.0 Mb/s 5745 MHz 11a -34dBm signal [bit 22] Beacon () [6.0* 9.0 12.0* 18.0 24.0* 36.0 48.0 54.0 Mbit] IBSS CH: 149, PRIVACY (MESHID: 11s-mesh-network)
    `
2. Using comma 
    `
12:20:37.867811 9526800862us tsft 6.0 Mb/s 5745 MHz 11a -34dBm signal [bit 22] Beacon () [6.0* 9.0 12.0* 18.0 24.0* 36.0 48.0 54.0 Mbit] IBSS CH: 149, PRIVACY, MESHID: 11s-mesh-network    
    `

Reply:
I prefer comma.  Can meshid have a comma in it?  Then "11s-mesh-network" might be best.

Reply:
@mcr, thanks a lot for your suggestion. Yes, the comma and even double quotes are allowed in Mesh ID and I didn't think about this. The Mesh ID field is exactly similar to the SSID field with respect to the allowed characters and length. So rather than enclosing the Mesh ID field with double quotes, we can stick to printing the Mesh ID field with parenthesis, so that its consistent with the way the SSID field is printed. we don't have to update the current patch in this PR, sorry for the confusion.